### PR TITLE
Issue #7339: document property type file

### PR DESCRIPTION
**Pull request by pbludov**
Issue #7339

New property type `file`.

Also, the documentation for all types that were called properties for some reason, even though they are types, has been fixed. Discovered by @rnveach at https://github.com/checkstyle/checkstyle/pull/7046#discussion_r336712228

### CODE REVIEW

File: src/xdocs/property_types.xml [hunk 1]
```diff
@@ -102,26 +102,33 @@
...
-        This property represents a URI. The string representation is parsed using a custom
+        This type represents a URI. The string representation is parsed using a custom
         routine to analyze what type of URI the string is.
         It can be a URL, regular file, or a resource path. It will try loading the path as
         a URL first, then as a file that must exist, and then finally as a resource on the
         classpath.
       </p>
     </section>
 
+    <section name="file">
```
Comment (romani) [inline src/xdocs/property_types.xml]: please move this update to separate commit that reference issue.
All other changes should be in separate commit, yuo can put "doc:" prefix for it.

Comment (pbludov) [reply src/xdocs/property_types.xml]: done
